akka-actor-tests/src/test/scala/akka/actor/CoordinatedShutdownSpec.scala

```diff
@@ -0,0 +1,133 @@
+/**
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.actor
+
+import scala.concurrent.Await
+import scala.concurrent.Future
+
+import akka.Done
+import akka.testkit.AkkaSpec
+import com.typesafe.config.ConfigFactory
+
+object CoordinatedShutdownSpec {
+
+}
+
+class CoordinatedShutdownSpec extends AkkaSpec {
+
+  def extSys = system.asInstanceOf[ExtendedActorSystem]
+
+  "CoordinatedShutdown" must {
+
+    "sort phases in topolgical order" in {
+      CoordinatedShutdown.topologicalSort(Map.empty) should ===(Nil)
+
+      CoordinatedShutdown.topologicalSort(Map(
+        "a" → Set.empty)) should ===(List("a"))
+
+      CoordinatedShutdown.topologicalSort(Map(
+        "b" → Set("a"))) should ===(List("a", "b"))
+
+      CoordinatedShutdown.topologicalSort(Map(
+        "c" → Set("a"), "b" → Set("a"))).head should ===("a") // b, c can be in any order
```

Comment:
shouldn't you also assert that "a" and "b" is also part of the list? Then this fully covers the case.